opm/simulators/flow/SimulatorFullyImplicitBlackoil.hpp

```diff
@@ -358,7 +358,7 @@ class SimulatorFullyImplicitBlackoil : private SerializableSim
                                         wmatcher,
                                         this->wellModel_().wellOpenTimes(),
                                         this->wellModel_().wellCloseTimes(),
-                                        [this, reportStep, schedule, timeStep](const std::string& name)
+                                        [reportStep, schedule, timeStep](const std::string& name)
```

Comment:
If you're changing this anyway, you might as well make it
```C++
[timeStep, &wg_events = schedule[reportStep].wellgroup_events()](const std::string& name) { ... }
```
instead.  That way it's clear we don't copy the full `Schedule` object into the lambda.

Reply:
wow, i seriously botched that lambda.

Reply:
> Capturing a reference-type-variable by value still does not do a copy, it seems.

That's true.  It's a special provision for lambda captures.  **However**: If you write
```C++
[sched = schedule](...) { ... }
```
(with `schedule` being a reference) then `sched` is a fully independent copy of the object to which `schedule` refers.

Reply:
If we write such code anywhere (to explicitly force a copy) I hope we at least add a comment...